diff --git a/dist/modules/build/lib/bundler/config.cjs.js b/dist/modules/build/lib/bundler/config.cjs.js
--- a/dist/modules/build/lib/bundler/config.cjs.js
+++ b/dist/modules/build/lib/bundler/config.cjs.js
@@ -95,10 +95,4 @@
-    const refreshOptions = {
-      overlay: {
-        sockProtocol: "ws",
-        sockHost: host,
-        sockPort: port
-      }
-    };
+    const refreshOptions = {};
     if (webpack) {
       const ReactRefreshPlugin = require("@pmmmwh/react-refresh-webpack-plugin");
       plugins.push(new ReactRefreshPlugin(refreshOptions));
diff --git a/dist/modules/build/lib/bundler/server.cjs.js b/dist/modules/build/lib/bundler/server.cjs.js
--- a/dist/modules/build/lib/bundler/server.cjs.js
+++ b/dist/modules/build/lib/bundler/server.cjs.js
@@ -164,13 +164,14 @@
-      host,
+      host: "0.0.0.0",
       port,
       proxy: targetPkg.proxy,
       // When the dev server is behind a proxy, the host and public hostname differ
-      allowedHosts: [url.hostname],
+      allowedHosts: "all",
       client: {
-        webSocketURL: { hostname: host, port }
+        webSocketURL: { hostname: "0.0.0.0", port: 0 }
       },
       headers: {
         "Access-Control-Allow-Origin": "*",
         "Access-Control-Allow-Methods": "GET, OPTIONS",
-        "Access-Control-Allow-Headers": "X-Requested-With, content-type, Authorization"
+        "Access-Control-Allow-Headers": "X-Requested-With, content-type, Authorization",
+        "Cache-Control": "no-store"
       }
